Automattic/jetpack
# [RNMobile] Fix resize video after upload

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/5603
Depends on: https://github.com/WordPress/gutenberg/pull/49929
The VP api will return a preview with default dimensions before it's fully processed the video. As a result the video will render cropped after uploading.  It can take up to 4 seconds before the correct video dimensions are available.

This PR will attempt to fetch the completed preview.

## Proposed changes:

* Adds a retry flow to request the video preview until its ready to render

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:





1. Add a VideoPress block and choose a video that does not have a 16:9 aspect ratio e.g. something recorded in portrait.
2. After a few seconds the video should appear with the correct aspect ratio.
3. Save and reload the post
4. The video should load with the correct aspect ratio with a delay.
